docs/releasing.md

```diff
@@ -0,0 +1,122 @@
+# Releasing a new Bats version
+
+These notes reflect the current process. There's a lot more we could do, in
+terms of automation and expanding the number of platforms to which we formally
+release (see #103).
+
+## Update docs/CHANGELOG.md
+
+Create a new entry at the top of `docs/CHANGELOG.md` that enumerates the
+significant updates to the new version.
+
+## Bumping the version number
+
+Bump the version numbers in the following files:
+
+- .appveyor.yml
+- contrib/rpm/bats.spec
+- libexec/bats-core/bats
+- package.json
+
+Commit these changes (including the `docs/CHANGELOG.md` changes) in a commit
+with the message `Bats <VERSION>`, where `<VERSION>` is the new version number.
+
+Create a new signed, annotated tag with:
+
+```bash
+$ git tag -a -s <VERSION>`
+```
+
+Include the `docs/CHANGELOG.md` notes corresponding to the new version as the
+tag annotation, except the first line should be: `Bats <VERSION> - YYYY-MM-DD`
+and any Markdown headings should become plain text, e.g.:
+
+```md
+### Added
+```
+
+should become:
+
+```md
+Added:
+```
+
+## Create a GitHub release
+
+Push the new version commit and tag to GitHub via the following (yes, it must be
+two separate commands):
+
+```bash
+# Push the version commit to `master`:
+$ git push
+
+# Push the version tag:
+$ git push --tags
```

Comment:
These two commands can just be single command: `git push --follow-tags`

Reply:
Updated. As I just said in the new commit comment, _this_ is why we have code
reviews: I'm learning useful little bits like this all the time!